File: CS/CustomParameterEditorsWPF/ViewModel.cs
using System.Collections.Generic;
using CustomParameterEditorsWPF.Reports;
using DevExpress.Mvvm.DataAnnotations;
using DevExpress.Xpf.Printing;
// ...

namespace CustomParameterEditorsWPF {
    [POCOViewModel]
    public class ViewModel {
        XtraReportPreviewModel model;
        public virtual Categories Categories { get; set; }

        public XtraReportPreviewModel Model {
            get {
                return model ?? (model = CreateReportModel());
            }
        }

        XtraReportPreviewModel CreateReportModel() {
            var model = new XtraReportPreviewModel();
            var report = new XtraReport1();
            model.Report = report;
            Categories = new Categories();
            model.ParametersModel.Validate += ParametersModel_Validate;
            report.CreateDocument();
            return model;
        }

        void ParametersModel_Validate(object sender, DevExpress.Xpf.Printing.Parameters.Models.ValidateParameterEventArgs e) {
            if(e.ParameterModel.Name == "catNames") {
                IList<object> categories = (IList<object>)e.ParameterModel.Value;
                if(categories.Count == 0) {
                    e.Error = "No categories selected";
                }
            }
        }
    }
}

File: VB/CustomParameterEditorsWPF/ViewModel.vb
Imports System.Collections.Generic
Imports CustomParameterEditorsWPF.Reports
Imports DevExpress.Mvvm.DataAnnotations
Imports DevExpress.Xpf.Printing

' ...
Namespace CustomParameterEditorsWPF

    <POCOViewModel>
    Public Class ViewModel

        Private modelField As XtraReportPreviewModel

        Public Overridable Property Categories As Categories

        Public ReadOnly Property Model As XtraReportPreviewModel
            Get
                Return If(modelField, Function()
                    modelField = CreateReportModel()
                    Return modelField
                End Function())
            End Get
        End Property

        Private Function CreateReportModel() As XtraReportPreviewModel
            Dim model = New XtraReportPreviewModel()
            Dim report = New XtraReport1()
            model.Report = report
            Categories = New Categories()
            AddHandler model.ParametersModel.Validate, AddressOf ParametersModel_Validate
            report.CreateDocument()
            Return model
        End Function

        Private Sub ParametersModel_Validate(ByVal sender As Object, ByVal e As Parameters.Models.ValidateParameterEventArgs)
            If Equals(e.ParameterModel.Name, "catNames") Then
                Dim categories As IList(Of Object) = CType(e.ParameterModel.Value, IList(Of Object))
                If categories.Count = 0 Then
                    e.Error = "No categories selected"
                End If
            End If
        End Sub
    End Class
End Namespace
